crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -75,13 +75,19 @@ def current_step_state
       progress[:steps][current_step] || {}
     end
 
-    def start_step
+    # 'step' is name of the step user wants to start.
+    # If `out_of_order` is true, the step could be executed out of order
+    def start_step(step_name, out_of_order = false)
```

Comment:
this would require changing all current uses of `start_step` to include the step, right? there's no backward compatibility with current "argumentless" call.